fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/Multiply.java

```diff
@@ -52,7 +53,12 @@ public DecimalV3Type getDataTypeForDecimalV3(DecimalV3Type t1, DecimalV3Type t2)
         int retPercision = t1.getPrecision() + t2.getPrecision();
         int retScale = t1.getScale() + t2.getScale();
         if (retPercision > DecimalV3Type.MAX_DECIMAL128_PRECISION) {
-            retPercision = DecimalV3Type.MAX_DECIMAL128_PRECISION;
+            boolean enableDecimal256 = ConnectContext.get().getSessionVariable().enableDecimal256();
+            if (enableDecimal256) {
+                retPercision = DecimalV3Type.MAX_DECIMAL256_PRECISION;
+            } else {
+                retPercision = DecimalV3Type.MAX_DECIMAL128_PRECISION;
+            }
```

Comment:
this if statement means upper round precision if result precision larger than max precision for decimal type. so u should change ret precision to 256 only if enableDecimal256 and precision > MAX_DECIMAL256_PRECISION. if enableDecimal256 is false, u should keep original logic